lib/declaration.ts

```diff
@@ -773,24 +775,33 @@ export default class DeclarationTranspiler extends base.TranspilerBase {
    */
   private visitProperty(
       decl: ts.PropertyDeclaration|ts.ParameterDeclaration, isParameter?: boolean) {
+    const hasValidName =
+        !ts.isStringLiteral(decl.name) || base.isValidDartIdentifier(decl.name.text);
     const isStatic = base.isStatic(decl);
-    this.emit('external');
-    if (isStatic) this.emit('static');
-    this.visit(decl.type);
-    this.emit('get');
-    this.visitName(decl.name);
-    this.emitNoSpace(';');
 
-    if (!base.isReadonly(decl)) {
+    // TODO(derekx): Properties with names that contain special characters are currently ignored by
+    // commenting them out. Determine a way to rename these properties in the future.
+    this.maybeWrapInCodeComment(/* shouldWrap */ !hasValidName, /* newLine */ true, () => {
       this.emit('external');
       if (isStatic) this.emit('static');
-      this.emit('set');
-      this.visitName(decl.name);
-      this.emitNoSpace('(');
       this.visit(decl.type);
-      this.emit('v');
-      this.emitNoSpace(')');
+      this.emit('get');
+      this.visitName(decl.name);
       this.emitNoSpace(';');
+    });
+
+    if (!base.isReadonly(decl)) {
+      this.maybeWrapInCodeComment(/* shouldWrap */ !hasValidName, /* newLine */ true, () => {
```

Comment:
using comments for the parameter names seems like a bit of an antipattern. perhaps use named parameters?

Reply:
Fixed.